File: haskell solutions/problem_16.hs
--Program finding the sum of the digits of a power of 2

power_of_two::Integer -> Integer
power_of_two number = 2^number

sum_of_digits::Integer -> Int
sum_of_digits number = 
    let list_digits = map (\x -> read [x] :: Int) (show number)
    in sum list_digits

main::IO()
main = do 
    let result = sum_of_digits (power_of_two (toInteger 1000))
    print(result)

File: python solutions/problem_16.py
# Program finding the sum of the digits of a power of 2


def power_of_two(power):
    return pow(2, power)


def sum_of_digits(number):
    string_number = str(number)
    sum_digits = 0
    for digit in string_number:
        sum_digits += int(digit)

    return sum_digits


if __name__ == "__main__":
    print(sum_of_digits(power_of_two(1000)))
